aiogram/utils/text_decorations.py

```diff
@@ -74,15 +74,15 @@ def _unparse_entities(
         length = length or len(text)
 
         for index, entity in enumerate(entities):
-            if entity.offset < offset:
+            if entity.offset * 2 < offset:
                 continue
-            if entity.offset > offset:
-                yield self.quote(text[offset : entity.offset])
-            start = entity.offset
-            offset = entity.offset + entity.length
+            if entity.offset * 2> offset:
+                yield self.quote(text[offset : entity.offset * 2].decode('utf-16-le'))
+            start = entity.offset * 2
+            offset = entity.offset * 2 + entity.length * 2
```

Comment:
Very small optimization:
```suggestion
            offset = (entity.offset + entity.length) * 2
```

Or maybe calculate doubled size once and then use it

For example:
```python3
for index, entity in enumerate(entities):
    entity_length = entity.length * 2
    entity_offset = entity.offset * 2


    if entity_offset < offset:
        continue
    ...
```

Reply:
I was trying to do that way, but there is recursive call of _unparse_entities, so it may lead to unexpected results for nested entities. I didn't have time to check the behavior and tine this index magic, so I decided to move decode/encode outside and pass the bytes. That's why I called the solution "raw" :)

Reply:
Hm.. Just try it and if this way is hard leave it as is 🙂

Reply:
And with recursive call, hmm, encoding outside `_unparse_entities` make sense..